test/services/companyProfileService.unit.ts

```diff
@@ -0,0 +1,69 @@
+import { Request } from "express";
+import { createOAuthApiClient } from "../../src/services/apiClientService";
+import { getCompanyProfile } from "../../src/services/companyProfileService";
+import { COMPANY_NUMBER, badRequestSDKResource, missingSDKResource, validCompanyProfile, validSDKResource } from "../mocks/companyProfile.mock";
+
+jest.mock("@companieshouse/api-sdk-node");
+jest.mock("../../src/services/apiClientService");
+
+const mockGetCompanyProfile = jest.fn();
+const mockCreateOAuthApiClient = createOAuthApiClient as jest.Mock;
+mockCreateOAuthApiClient.mockReturnValue({
+    companyProfile: {
+        getCompanyProfile: mockGetCompanyProfile
+    }
+});
+
+describe("CompanyProfileService", () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    afterEach(() => {
+        expect(mockCreateOAuthApiClient).toHaveBeenCalledTimes(1);
+        expect(mockGetCompanyProfile).toHaveBeenCalledTimes(1);
+    });
+
+    describe("getPscVerification", () => {
+        it("should retrieve the company profile resource", async () => {
```

Comment:
Add a test scenario: getCompanyProfile() response is an ApiErrorResponse object with "errors" property.

Reply:
I have added that additional scenario